frontend/javascripts/oxalis/controller.js

```diff
@@ -281,22 +292,54 @@ class Controller extends React.PureComponent<PropsWithRouter, State> {
 
   render() {
     const status = this.props.controllerStatus;
-    const { user, viewMode } = this.props;
-    const { gotUnhandledError } = this.state;
+    const { user, viewMode, initialCommandType } = this.props;
+    const { gotUnhandledError, usersOrganizations } = this.state;
+    let suggestedOrganization = null;
+    if (
+      user != null &&
+      initialCommandType.type === ControlModeEnum.VIEW &&
+      initialCommandType.owningOrganization !== user.organization
+    ) {
+      // Retrieving organization name from url. This only works in view mode.
+      const owningOrganizationName = location.pathname.split("/")[2];
+      suggestedOrganization = usersOrganizations.find(
+        orga => orga.name === owningOrganizationName && orga.name !== user.organization,
+      );
```

Comment:
Hm, I think, it would be very good to also support the scenario where an annotation link points to another orga. I'm wondering whether the back-end could check whether the user is member of the corresponding orga? Then, the back-end could reply with the organization id (since this is only done, if the user can switch to that orga, this is not leaking any private information).
The benefits of that would be:
- no parsing of the url to extract the organization
- support for annotation links
- no need to fetch the user's organizations async (currently, this can lead to a weird UI, where the 404 is already shown, but the "auto-switch possibility" is still fetched)

@fm3 Do you think it makes sense to do this on the back-end?

Reply:
I started implementing an `accessibleBySwitching` route. It is not finished yet, I will continue tomorrow.